### PR TITLE
treewide: unify indentical definitions of polkit_agent_open_if_enabled()

### PR DESCRIPTION
**Pull request by keszybz**
Follows the same pattern as pager_open() now.

### CODE REVIEW

**Comment by yuwata** [comment]
indentical -> identical

**Comment by keszybz** [comment]
```
 65/245 test-async                              TIMEOUT 49.29 s
```
Strange.

**Comment by poettering** [comment]
so xenial-amd64 fails, but I can't make sense of it?

lgtm otherwise

**Comment by keszybz** [comment]
xenial-amd64 fails on the standard qemu crash, might be caused by #7171 or not. Either way it's certainly not related to this PR.

xenial-s390x failed on test-async. This already happened in some other PR. It's very strange, but maybe the machine is overloaded and stalls for one second. Maybe we should extend the timeout...

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -268,20 +268,13 @@ static void ask_password_agent_open_if_enabled(void) {
...
-static void polkit_agent_open_if_enabled(void) {
-
+static void polkit_agent_open_maybe(void) {
         /* Open the polkit agent as a child process if necessary */
 
-        if (!arg_ask_password)
-                return;
-
         if (arg_scope != UNIT_FILE_SYSTEM)
                 return;
```
Comment (tixxdz) [inline src/systemctl/systemctl.c:276]: Just dumping my thoughts, does it make sense to check other scopes like global... ?

**Comment by tixxdz** [comment]
@keszybz one comment not sure if it makes sense, didn't check what's logic behind, otherwise all call sites lgtm, thanks!

File: src/systemctl/systemctl.c (hunk 1 above)
Comment (keszybz) [reply src/systemctl/systemctl.c:276]: No. Global scope is done through filesystem operations, not over bus, and pid1 is not involved in this at all.

**Pull request merged**